pyanaconda/modules/payloads/source/cdrom/initialization.py

```diff
@@ -0,0 +1,115 @@
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from blivet.arch import get_arch
+
+from pyanaconda.modules.common.constants.objects import DEVICE_TREE
+from pyanaconda.modules.common.constants.services import STORAGE
+from pyanaconda.modules.common.errors.payload import SourceSetupError
+from pyanaconda.modules.common.errors.storage import MountFilesystemError
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.structures.storage import DeviceData
+from pyanaconda.payload.utils import mount, unmount
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+__all__ = ["SetUpCdromSourceTask", "TearDownCdromSourceTask"]
+
+
+class TearDownCdromSourceTask(Task):
+    """Task to teardown installation source."""
+
+    def __init__(self, target_mount):
+        super().__init__()
+        self._target_mount = target_mount
+
+    @property
+    def name(self):
+        return "Tear down CD-ROM Installation Source"
+
+    def run(self):
+        """Run live installation source un-setup."""
+        log.debug("Unmounting CD-ROM installation source")
+        unmount(self._target_mount)
+
+
+class SetUpCdromSourceTask(Task):
+    """Task to setup installation source."""
+
+    def __init__(self, target_mount):
+        super().__init__()
+        self._target_mount = target_mount
+        self._device = ""
+
+    @property
+    def name(self):
+        return "Set up CD-ROM Installation Source"
+
+    def run(self):
+        """Run CD-ROM installation source setup."""
+        log.debug("Trying to detect CD-ROM automatically")
+
+        if self._device:
+            raise SourceSetupError("CD-ROM already mounted")
+
+        device_tree = STORAGE.get_proxy(DEVICE_TREE)
+
+        for dev_name in device_tree.FindOpticalMedia():
+            try:
+                device_data = DeviceData.from_structure(device_tree.GetDeviceData(dev_name))
+                mount(device_data.path, self._target_mount, "iso9660", "ro")
+                # or...
+                # device_tree.MountDevice(dev_name, self._target_mount)
+                # RO is implied in blivet.tasks.fsmount.Iso9660FSMount that is eventually called
+            except MountFilesystemError:
+                continue
+
+            if _is_valid_install_disk(self._target_mount):
+                self._device = dev_name
+                # refresh device_tree ?
+                break
+            else:
+                unmount(self._target_mount)
+                # or...
+                # device_tree.UnmountDevice(dev_name, self._target_mount)
+
+        if not self._device:
+            raise SourceSetupError("Found no CD-ROM")
+
+
+def _is_valid_install_disk(mount_dir):
```

Comment:
Please, don't duplicate the code. When it is possible to use an existing function, use it. I would create a new file `utils.py` in `pyanaconda/modules/payloads/source/` or `pyanaconda/modules/payloads/source/cdrom`, move `verifyMedia` there, rename it and clean it up. It doesn't matter that anaconda will temporarily import something from a DBus module.

Reply:
Changed according to your description.